netdissect.h

```diff
@@ -391,7 +391,7 @@ nd_trunc_longjmp(netdissect_options *ndo)
  * Number of bytes remaining in the captured data, starting at the
  * byte pointed to by the argument.
  */
-#define ND_BYTES_AVAILABLE_AFTER(p) ND_BYTES_BETWEEN(ndo->ndo_snapend, (p))
+#define ND_BYTES_AVAILABLE_AFTER(p) ((const u_char *)(p) >= ndo->ndo_snapend || (const u_char *)(p) < ndo->ndo_packetp ? 0 : (u_int)(ndo->ndo_snapend - (const u_char *)(p)))
```

Comment:
Or maybe

    #define ND_BYTES_AVAILABLE_AFTER(p) ((const u_char *)(p) >= ndo->ndo_snapend || (const u_char *)(p) < ndo->ndo_packetp ? 0 : ND_BYTES_BETWEEN(ndo->ndo_snapend, (p)))

Reply:
Or perhaps update first `ND_BYTES_BETWEEN`.
In f9c2c905b118b69a0b102549c1b25cca871947b5, message we have:
```
Add an ND_BYTES_BETWEEN() macro that computes how many bytes are 
present, starting at the second argument and running up to (but not 
including) the first argument, and returns that as a u_int [...]
```
If p1 >= p2: OK.
If p1 < p2: negative -> big value via the u_int cast. Should be zero (no bytes present).

Reply:
Someting like:
`#define ND_BYTES_BETWEEN(p1, p2) ((const u_char *)(p2) >= (const u_char *)(p1) ? 0 : (u_int)((const u_char *)(p1) - (const u_char *)(p2)))`
or cast with `uint8_t *`if this really make a difference.

Reply:
I think the purpose was 1.

Reply:
Yes, the purpose was 1.  The goal was to have a macro to find out how many bytes there were from p2 to p1 (so p1 >= p2) and return the result as a `u_int`, with an explicit cast to squelch any "that's a 64-bit quantity and you're throwing away the upper 32 bits" warnings that you get (either from Apple's compilers, which can be told to warn about 64->32 narrowings, or from Visual C/++, which can be told to warn about *all* narrowings).

Reply:
Ok. I have reverse, in the first commit, the order of the arguments because it seems to me more natural to go from p1 to p2 (no need to keep the order from the old SMB's PTR_DIFF()).